src/Commands/PullInsights.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tipoff\Reviews\Commands;
+
+use Carbon\Carbon;
+use Google_Client;
+use Google_Service_MyBusiness;
+use Google_Service_MyBusiness_BasicMetricsRequest;
+use Google_Service_MyBusiness_MetricRequest;
+use Google_Service_MyBusiness_ReportLocationInsightsRequest;
+use Google_Service_MyBusiness_TimeRange;
+use Illuminate\Console\Command;
+use Tipoff\Locations\Models\Location;
+use Tipoff\Reviews\Models\Insight;
+use Tipoff\Reviews\Models\Key;
+
+class PullInsights extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'pull:insights';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Pull the GMB Insights for each location';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        // @todo Needs Refactoring
+        // I'm refreshing the stats from the previous few days since the data appears to be delayed. If yesterday's stats become accurate as of 7am, then can remove this.
+        $days = [45, 44, 43, 42, 41, 40, 39, 38, 37, 36, 35, 34, 33, 32, 31, 30, 29, 28, 27, 26, 25, 24, 23, 22, 21, 20, 19, 18, 17, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1];
+
+        foreach ($days as $day) {
+            $locations = $locations = Location::whereNotNull('gmb_location')->get();
+
+            foreach ($locations as $location) {
+                $client = new Google_Client();
+                // @todo Needs Refactoring
+                $client->setAuthConfig(resource_path('client_secret.json'));
+                $client->addScope(['https://www.googleapis.com/auth/business.manage']);
+                $client->setAccessType('offline');
+
+                $token = json_decode(Key::where('slug', 'gmb-token')->first()->value, true);
+                $client->setAccessToken($token);
+
+                if ($client->isAccessTokenExpired()) {
+                    $client->refreshToken(array_search('refresh_token', $token));
+                    $newtoken = $client->fetchAccessTokenWithRefreshToken($client->getRefreshToken());
+                    $savetoken = Key::updateOrCreate(
+                        ['slug' => 'gmb-token'],
+                        ['value' => json_encode($newtoken)]
+                    );
+                }
+
+                // @todo Needs Refactoring
+                if ($location->corporate == 0) {
+                    $account = 'accounts/116666006358174413896';
```

Comment:
How do we want to handle account numbers being passed in? Should this live in the config file?

Reply:
There is now an gmb_account field on the locations table